client/lib/paths/login/index.js

```diff
@@ -34,7 +36,7 @@ export function login( {
 		url += '/social-connect';
 	} else if ( isJetpack ) {
 		url += '/jetpack';
-	} else if ( isGutenboarding ) {
+	} else if ( isGutenboarding || isReskinned ) {
```

Comment:
Is it possible to remove the `isGutenboarding` flag and unify both `is*` flags into one?

The `isReskinned` naming is also a bit ambiguous: at different places, we use the "reskinned" and "white" names for the same thing.

The `isGutenboarding` condition should also be modified in the `if` branch a few lines above where we're construcing the `'/new/' + twoFactorAuthType` URL.

Reply:
> Is it possible to remove the isGutenboarding flag and unify both is* flags into one?

It should be possible and will be more clean/elegant. 👍 

> The `isReskinned` naming is also a bit ambiguous: at different places, we use the "reskinned" and "white" names for the same thing.

Makes sense, and yeah there's definitely more overhead introduced by having both. The intention with the `isWhiteLogin` flag in this PR was to align it closer with the `is-white-login` styles. "reskinned" is used quite heavily, so if it makes sense I'd opt for making more use of that than the opposite, in general. We are obviously not going to make a big rename/refactor through and through in this PR. Any suggestions though for this case at least?

Reply:
As a rule of thumb for naming things, "absolute" names like "white" are better than relative ones like "reskinned" or "new". Because the latter require some extra context (reskinned from what to what?) that becomes irrelevant over time.

It's OK to merge this PR as is if cleanups would have some snowball effect inflating the patch too much. There can be a separate follow-up.

Reply:
Yeah, good point(s). 

Hopefully all of this gets cleaned up soon. I understand there is a bit of decision-making coming up to deprecate Gutenboarding and the old/blue flow - which for all I know would/should discuss the shape (name & form) of what remains. It sort of makes sense to introduce a bit more overhead here and ship the functionality (see also the styles and all the commenting I'm adding 🤷 ), and let a dedicated refactor bring everything together when it's time.

Reply:
> As a rule of thumb for naming things, "absolute" names like "white" are better than relative ones like "reskinned" or "new". Because the latter require some extra context (reskinned from what to what?) that becomes irrelevant over time.

This one addressed in https://github.com/Automattic/wp-calypso/pull/54646/commits/4be7eb0f72a99fd93c7e8f8ffc5a52a5416d4243